homeassistant/components/forked_daapd/media_player.py

```diff
@@ -794,26 +794,29 @@ async def _update(self, update_types):
             "queue" in update_types
         ):  # update queue, queue before player for async_play_media
             queue = await self._api.get_request("queue")
-            update_events["queue"] = asyncio.Event()
-            async_dispatcher_send(
-                self.hass,
-                SIGNAL_UPDATE_QUEUE.format(self._entry_id),
-                queue,
-                update_events["queue"],
-            )
+            if queue:
+                update_events["queue"] = asyncio.Event()
+                async_dispatcher_send(
+                    self.hass,
+                    SIGNAL_UPDATE_QUEUE.format(self._entry_id),
+                    queue,
+                    update_events["queue"],
+                )
         # order of below don't matter
         if not {"outputs", "volume"}.isdisjoint(update_types):  # update outputs
-            outputs = (await self._api.get_request("outputs"))["outputs"]
-            update_events[
-                "outputs"
-            ] = asyncio.Event()  # only for master, zones should ignore
-            async_dispatcher_send(
-                self.hass,
-                SIGNAL_UPDATE_OUTPUTS.format(self._entry_id),
-                outputs,
-                update_events["outputs"],
-            )
-            self._add_zones(outputs)
+            outputs = await self._api.get_request("outputs")
+            if outputs:
+                outputs = outputs["outputs"]
```

Comment:
Is this changing due to a change in the library?

Reply:
So why do we change outputs here?

Reply:
The previous version had the dictionary access in the same line as the function call, while now it checks the return value of the function first and has the dictionary access/reassignment after.

Reply:
Right, thanks.